backend/src/main/java/com/redhat/cloud/notifications/routers/EmailEndpointMigrationService.java

```diff
@@ -0,0 +1,150 @@
+                            .onItem().transformToUni(endpoints -> {
+                                if (endpoints.size() > 1) {
+                                    Endpoint defaultEmailEndpoint = endpoints.remove(0);
+                                    LOGGER.debugf("[NOTIF-242] Account [%s] has [%d] EmailEndpoints. Using [%s] as the default", accountId, endpoints.size() + 1, defaultEmailEndpoint.getId());
+                                    Set<UUID> endpointIds = endpoints.stream().map(Endpoint::getId).collect(Collectors.toSet());
+
+                                    return session.createQuery("FROM BehaviorGroupAction bga WHERE bga.id.endpointId IN (:endpointIds)", BehaviorGroupAction.class)
+                                            .setParameter("endpointIds", endpointIds)
+                                            .getResultList()
+                                            .onItem().transformToMulti(Multi.createFrom()::iterable)
+                                            .invoke(behaviorGroupAction -> {
+                                                LOGGER.debugf(
+                                                        "[NOTIF-242] Updating BehaviorGroup [%s] to use default EmailEndpoint [%s]",
+                                                        behaviorGroupAction.getId().behaviorGroupId,
+                                                        defaultEmailEndpoint.getId()
+                                                );
+                                            })
+                                            .onItem().transformToUniAndConcatenate(behaviorGroupAction -> session.remove(behaviorGroupAction)
+                                                    .onItem().transformToUni(_unused -> {
+                                                        BehaviorGroupActionId bgaId = new BehaviorGroupActionId();
+                                                        bgaId.endpointId = defaultEmailEndpoint.getId();
+                                                        bgaId.behaviorGroupId = behaviorGroupAction.getId().behaviorGroupId;
+
+                                                        return session.find(BehaviorGroupAction.class, bgaId)
+                                                                .onItem().transformToUni(bgaFound -> {
+                                                                    if (bgaFound == null) {
+                                                                        BehaviorGroupAction replacement = new BehaviorGroupAction(
+                                                                                behaviorGroupAction.getBehaviorGroup(),
+                                                                                defaultEmailEndpoint
+                                                                        );
+                                                                        replacement.setPosition(behaviorGroupAction.getPosition());
```

Comment:
IIUC, some positions may be empty after the migration, but I don't think that's a problem.

Reply:
Yes, some positions will be like 1, 2, 5, 8. But that shouldn't be a problem.